src/NppFTP.cpp

```diff
@@ -305,57 +305,55 @@ void NppFTP::AttemptToAutoConnect(const TCHAR* path)
 	if (!path)
 		return;
 
-	FTPProfile* profile = nullptr;
+	FTPProfile* profile = 0;	// make nullptr
 
-	// we're going to parse the current username from the file path
+	// we're going to parse the current username from the file path, let's find the objects around at @ symbol
 	TCHAR* charScanner = const_cast<TCHAR*>(path);
+	int charCounter = 0;
 	int atSymbolIndex = 0;
 	int previousBackslashIndex = 0;
-	while (*charScanner != '\0')
-	{
-		if (charScanner[atSymbolIndex] == (TCHAR)'\\')
-			previousBackslashIndex = atSymbolIndex + 1;
-		if (charScanner[atSymbolIndex] == (TCHAR)'@')
-			break;
-		atSymbolIndex++;
+	int nextBackslashIndex = 0;
+	while (charScanner[charCounter] != '\0')
+	{	
+		if (charScanner[charCounter] == (TCHAR)'@')
+			atSymbolIndex = charCounter;
+		if (atSymbolIndex > 0 && (charScanner[charCounter] == (TCHAR)'\\' || charScanner[charCounter] == (TCHAR)'/'))
+		{
+			nextBackslashIndex = charCounter;
+			break;	// prevents resetting of the previous backslash index by breaking out here
+		}
+		if (atSymbolIndex == 0 && (charScanner[charCounter] == (TCHAR)'\\' || charScanner[charCounter] == (TCHAR)'/'))
+			previousBackslashIndex = charCounter + 1;
+		charCounter++;
 	}
 
-	// we store that parsed username into the variable username below
-	if (atSymbolIndex>0)
+	// we store that parsed username into the variable username below if we found an @ symbol
+	if (previousBackslashIndex >0 && atSymbolIndex > previousBackslashIndex && nextBackslashIndex > atSymbolIndex)
 	{
-		TCHAR username[FILENAME_MAX];	// assume username+path is smaller than largest allowed filename
-		memset(username, '\0', FILENAME_MAX);
-		int nameLength = (atSymbolIndex - previousBackslashIndex - 1);
+		TCHAR username[FILENAME_MAX*2];	// assume username+path is smaller than largest allowed filename, x2 because TCHAR is 2 bytes
+		TCHAR hostname[FILENAME_MAX * 2];	// assume username+path is smaller than largest allowed filename, x2 because TCHAR is 2 bytes
+		memset(username, '\0', FILENAME_MAX*2);
+		memset(hostname, '\0', FILENAME_MAX * 2);
+		int nameLength = (atSymbolIndex - previousBackslashIndex);
+		int hostnameLength = (nextBackslashIndex - atSymbolIndex - 1);
 		memcpy(username, &path[previousBackslashIndex], nameLength * 2);	// TCHAR is 2 bytes so we have to double the nameLength to be correct in byte-space

```

Comment:
use sizeof(TCHAR) instead of hardcoded 2 (for old ascii builds TCHAR would be just 1 byte)

Reply:
nice catch